### PR TITLE
ADD version 0.19.0 in py-gym recipe

### PR DESCRIPTION
**Pull request by emilioj**
In addition to v0.18.0, allow the installation of gym 0.19.0

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-gym/package.py [hunk 1]
```diff
@@ -16,6 +16,7 @@ class PyGym(PythonPackage):
     homepage = "https://github.com/openai/gym"
     pypi = "gym/0.18.0.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-gym/package.py]:
```suggestion
    pypi = "gym/gym-0.18.0.tar.gz"
```
No idea how this ever worked. I guess we've been relying on the downloads mirror.

File: var/spack/repos/builtin/packages/py-gym/package.py [hunk 2]
```diff
@@ -16,6 +16,7 @@ class PyGym(PythonPackage):
...
 
+    version('0.19.0', sha256="940069b983806e1ccc400fa6d47b4e34e462accf6a4fb0acb0a5e509ad0f502d")
     version("0.18.0", sha256="a0dcd25c1373f3938f4cb4565f74f434fba6faefb73a42d09c9dddd0c08af53e")
 
     depends_on("python@3.6:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-gym/package.py:22]: Dependencies have changed:
```python
    install_requires=[                                                                   
        "numpy>=1.18.0",                                                                 
        "cloudpickle>=1.2.0,<1.7.0",                                                     
    ],    
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]
It looks like there are much newer versions available if you want newer gym.

File: var/spack/repos/builtin/packages/py-gym/package.py (hunk 1 above)
Comment (emilioj) [reply var/spack/repos/builtin/packages/py-gym/package.py]: First of all: thanks for your time!

As a spack newbie I did not realize about the url issue :-/

I've added a second commit fixing this and the dependencies you mentioned in the other comment (my completely fault not to check that, sorry O:) ).

I've locally rebased my two commits, the commit I used for this PR and the new one fixing the issues and then I've forced a push in my develop branch... but I am not sure if that is the appropriate way of acting (I mean doing that previous rebase with the current state of the develop branch). :-?

**Comment by emilioj** [comment]
> It looks like there are much newer versions available if you want newer gym.

Sure, I know, of course. I've behaved completely selfishly updating the recipe only with the specific version I need for the software stack I use in a project I am involved O:)

I do not have time these days (I am fighting with a deadline, indeed) to check & try all more recent available versions, but I can do it in a couple of weeks.

File: var/spack/repos/builtin/packages/py-gym/package.py (hunk 2 above)
Comment (emilioj) [reply var/spack/repos/builtin/packages/py-gym/package.py:22]: I think I addressed the new dependencies in my second commit (see previous message about my -maybe wrong- rebase)

File: var/spack/repos/builtin/packages/py-gym/package.py [hunk 3]
```diff
@@ -14,16 +14,17 @@ class PyGym(PythonPackage):
...
+    depends_on("py-scipy", type=("build", "run"), when="@0.18.0")
+    depends_on("py-numpy@1.10.4:", type=("build", "run"), when="@0.18.0")
+    depends_on("py-numpy@1.18.0:", type=("build", "run"), when="@0.19.0")
     depends_on("py-pyglet@1.4.0:1.5.0", type=("build", "run"), when="@0.18.0")
-    depends_on("py-pyglet@1.4.0:1.5.15", type=("build", "run"), when="@0.18.1")
-    depends_on("pil@:8.2.0", type=("build", "run"), when="@0.18.1")
     depends_on("pil@:7.2.0", type=("build", "run"), when="@0.18.0")
-    depends_on("py-cloudpickle@1.2.0:1.6", type=("build", "run"))
+    depends_on("py-cloudpickle@1.2.0:1.6", type=("build", "run"), when="@0.18.0")
+    depends_on("py-cloudpickle@1.2.0:1.7.0", type=("build", "run"), when="@0.19.0")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-gym/package.py]:
```suggestion
    depends_on("py-cloudpickle@1.2.0:1.6", type=("build", "run"))
```
This dep hasn't changed. It's `<1.7`, which is `:1.6` in Spack (version ranges are inclusive in Spack).

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-gym/package.py (hunk 3 above)
Comment (emilioj) [reply var/spack/repos/builtin/packages/py-gym/package.py]: O:-)

I know how version ranges work... I do not know why I saw a difference in the dep version for 0.19...

Sorry!

File: var/spack/repos/builtin/packages/py-gym/package.py [hunk 4]
```diff
@@ -14,16 +14,16 @@ class PyGym(PythonPackage):
...
-    depends_on("py-numpy@1.10.4:", type=("build", "run"))
+    depends_on("py-scipy", type=("build", "run"), when="@0.18.0")
+    depends_on("py-numpy@1.10.4:", type=("build", "run"), when="@0.18.0")
+    depends_on("py-numpy@1.18.0:", type=("build", "run"), when="@0.19.0")
     depends_on("py-pyglet@1.4.0:1.5.0", type=("build", "run"), when="@0.18.0")
-    depends_on("py-pyglet@1.4.0:1.5.15", type=("build", "run"), when="@0.18.1")
-    depends_on("pil@:8.2.0", type=("build", "run"), when="@0.18.1")
     depends_on("pil@:7.2.0", type=("build", "run"), when="@0.18.0")
-    depends_on("py-cloudpickle@1.2.0:1.6", type=("build", "run"))
+    depends_on("py-cloudpickle@1.2.0:1.6", type=("build", "run"), when="@0.18.0")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-gym/package.py]:
```suggestion
    depends_on("py-cloudpickle@1.2.0:1.6", type=("build", "run"))
```

Comment (emilioj) [reply var/spack/repos/builtin/packages/py-gym/package.py]: So embarrassing... definitely I should pay more attention O:)

Thanks again for your time and your patience!

E

**Comment by adamjstewart** [comment]
@spackbot fix style

**Review by adamjstewart — DISMISSED** [review]

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**